actors/builtin/payment_channel/payment_channel_actor.go

```diff
@@ -0,0 +1,247 @@
+package payment_channel
+
+import (
+	"bytes"
+
+	addr "github.com/filecoin-project/go-address"
+	abi "github.com/filecoin-project/specs-actors/actors/abi"
+	big "github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	vmr "github.com/filecoin-project/specs-actors/actors/runtime"
+	indices "github.com/filecoin-project/specs-actors/actors/runtime/indices"
+	"github.com/filecoin-project/specs-actors/actors/serde"
+	"github.com/minio/blake2b-simd"
+)
+
+type PaymentChannelActor struct{}
+
+// type PaymentInfo struct {
+// 	PayChActor     address.Address
+// 	Payer          address.Address
+// 	ChannelMessage *cid.Cid
+
+// 	Vouchers []*types.SignedVoucher
+// }
+
+/////////////////
+// Constructor //
+/////////////////
+
+type PCAConstructorParams struct {
+	To addr.Address
+}
+
+func (pca *PaymentChannelActor) Constructor(rt vmr.Runtime, params *PCAConstructorParams) *vmr.EmptyReturn {
+
+	var st PaymentChannelActorState
+	rt.State().Transaction(&st, func() interface{} {
+		st.From = rt.ImmediateCaller()
+		st.To = params.To
+		st.LaneStates = make(map[int64]*LaneState)
+		return nil
+	})
+	return &vmr.EmptyReturn{}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Payment Channel state operations
+////////////////////////////////////////////////////////////////////////////////
+
+type PCAUpdateChannelStateParams struct {
+	Sv     SignedVoucher
+	Secret []byte
+	Proof  []byte
+}
+
+func hash(b []byte) []byte {
+	s := blake2b.Sum256(b)
+	return s[:]
+}
+
+type PaymentVerifyParams struct {
+	Extra []byte
+	Proof []byte
+}
+
+func (pca *PaymentChannelActor) UpdateChannelState(rt vmr.Runtime, params *PCAUpdateChannelStateParams) *vmr.EmptyReturn {
+
+	var st PaymentChannelActorState
+	rt.State().Transaction(&st, func() interface{} {
+
+		sv := params.Sv
+
+		vb, nerr := sv.SigningBytes()
+		if nerr != nil {
+			rt.Abort(1, "failed to serialize signedvoucher")
```

Comment:
Invalid exit codes, must be greater than `exitcode.FirstActorErrorCode`. Consider using those already defined in common.go where they are a good fit, but also absolutely ok to define others locally here, with symbolic constants.